### PR TITLE
[SPARK-31882][WEBUI] DAG-viz is not rendered correctly with pagination

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR fix an issue related to DAG-viz.
Because DAG-viz for a job fetches link urls for each stage from the stage table, rendering can fail with pagination.
You can reproduce this issue with the following operation.
```
sc.parallelize(1 to 10).map(value => (value ,value)).repartition(1).repartition(1).repartition(1).reduceByKey(_ + _).collect
```
And then, visit the corresponding job page.
There are 5 stages so show <5 stages in the paged table.
dag-rendering-issue1
dag-rendering-issue2

### Why are the changes needed?

This is a bug.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Newly added test case with following command.
`build/sbt -Dtest.default.exclude.tags= -Dspark.test.webdriver.chrome.driver=/path/to/chromedriver "testOnly org.apache.spark.ui.ChromeUISeleniumSuite -- -z SPARK-31882"`

### CODE REVIEW

**Review by srowen — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Looks okay. cc @gengliangwang FYI

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala [hunk 1]
```diff
@@ -292,6 +292,7 @@ private[spark] object UIUtils extends Logging {
     <html>
       <head>
         {commonHeaderNodes(request)}
+        <script>setAppBasePath('{activeTab.basePath}')</script>
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/ui/UIUtils.scala:295]: I think we can move this line to function `commonHeaderNodes`

**Review by gengliangwang — APPROVED** [review]
LGTM except one comment

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala (hunk 1 above)
Comment (sarutak) [reply core/src/main/scala/org/apache/spark/ui/UIUtils.scala:295]: At first, I considered the way you mention but to do so, we need to have an additional parameter in `commonHeaderNodes` to pass `activeTab` only for this purpose.
Which do you think is better?

Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/ui/UIUtils.scala:295]: I see. Let's keep it this way. The method `basicSparkPage` also calls the function `commonHeaderNodes` but there is no `activeTab` in it.

**Comment by gengliangwang** [comment]
@sarutak Does the bug exist in branch 3.0?

**Comment by sarutak** [comment]
@gengliangwang I've confirmed this bug exists in branch 3.0 too.

**Comment by gengliangwang** [comment]
Then, let's merge to both master and 3.0 :)

**Comment by sarutak** [comment]
Yeah, but #28627 isn't merged to `branch-3.0` so we need to back port it first or open another PR for `branch-3.0` to back port this change without the testcase.

**Comment by sarutak** [comment]
Anyway, I'll open a PR for backporting #28627.

**Comment by gengliangwang** [comment]
Awesome, let's merge this one after #28627 is backported 👍

**Comment by gengliangwang** [comment]
Thanks, merging to master/3.0

**Pull request closed without merging**